test/hotspot/gtest/nmt/test_vmatree.cpp

```diff
@@ -739,4 +750,78 @@ TEST_VM_F(NMTVMATreeTest, SummaryAccountingWhenUseFlagInplace) {
   diff = tree.uncommit_mapping(0, 50, rd2);
   EXPECT_EQ(0, diff.tag[NMTUtil::tag_to_index(mtTest)].reserve);
   EXPECT_EQ(-50, diff.tag[NMTUtil::tag_to_index(mtTest)].commit);
+}
+
+TEST_VM_F(NMTVMATreeTest, SeparateStacksForCommitAndReserve) {
+  using SIndex = NativeCallStackStorage::StackIndex;
+  SIndex si_1 = si[0];
+  SIndex si_2 = si[1];
+
+  VMATree::RegionData call_stack_1(si_1, mtTest);
+  VMATree::RegionData call_stack_2(si_2, mtNone);
+  auto expected =  [&](Tree& tree, int p, SIndex reserve_stack, SIndex commit_stack, int line_no = __LINE__) {
+    VMATree::VMATreap::Range r = tree.tree().find_enclosing_range(p);
+    const bool check_end_node = true;
+    const char* at_line = " at line: ";
+    if (reserve_stack >= 0) {
+      EXPECT_EQ(r.start->val().out.reserved_stack(), reserve_stack) << at_line << line_no;
+      EXPECT_EQ(r.end->val().in.reserved_stack(), reserve_stack) << at_line << line_no;
+    } else {
+      EXPECT_FALSE(r.start->val().out.has_reserved_stack()) << at_line << line_no;
+      EXPECT_FALSE(r.end->val().in.has_reserved_stack()) << at_line << line_no;
+    }
+    if (commit_stack >= 0) {
+      EXPECT_EQ(r.start->val().out.committed_stack(), commit_stack) << at_line << line_no;
+      EXPECT_EQ(r.end->val().in.committed_stack(), commit_stack) << at_line << line_no;
+    } else {
+      EXPECT_FALSE(r.start->val().out.has_committed_stack()) << at_line << line_no;
+      EXPECT_FALSE(r.end->val().in.has_committed_stack()) << at_line << line_no;
+    }
+  };
+
+  {
```

Comment:
I think that each test case should at least have a small comment summarizing the supposed goal of the test. Below is an example of what the could look like for the first test case. 

```
// Check committing into a reserved region inherits the call stacks
```